lib/linked-list.js

```diff
@@ -19,148 +20,232 @@ class LinkedList {
+        let count = 0;
+        let current = this.#head;
+        while (current) {
+            count ++
+            current = current.next
+            if (current == this.#head){
+                return count
+            }
+        }
+        return count   
     }
 
     /*
     method that returns the value at a given index in the linked list
     index count starts at 0
     returns nil if there are fewer nodes in the linked list than the index value
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n)
+    Space Complexity: O(1)
      */
     getAtIndex(index) {
-        throw new Error("This method hasn't been implemented yet!");
+        let count = 0;
+        let current = this.#head;
+        while (current) {
+            if (count === index) {
+                return current.value
+            } else {
+                count ++
+                current = current.next
+            }
+        }
+        return null
     }
 
     /*
     method that returns the value of the last node in the linked list
     returns nil if the linked list is empty
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n)
+    Space Complexity: O(1)
     */
     getLast() {
-        throw new Error("This method hasn't been implemented yet!");
+        let current = this.#head;
+        while (current.next) {
+            current = current.next
+        }
+        return current.value
     }
 
     /*
     method that inserts a given value as a new last node in the linked list
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n)
+    Space Complexity: O(1)
     */
     addLast(value) {
-        throw new Error("This method hasn't been implemented yet!");
+        if (!this.#head) {
+            this.#head = new Node(value)
+        } else {
+            let current = this.#head;
+            while (current.next) {
+                current = current.next
+            }
+            current.next = new Node(value)
+        }
     }
 
     /*
     method to return the max value in the linked list
     returns the data value and not the node
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n)
+    Space Complexity: O(1)
     */
     findMax() {
-        throw new Error("This method hasn't been implemented yet!");
+        if (!this.#head) {
+            return null
+        }
+        let max = this.#head.value
+        let current = this.#head;
+        while (current) {
+            (current.value>max)? (max = current.value):
+            current = current.next
+        }
+        return max;
     }
 
     /*
     method to delete the first node found with specified value
-    Time Complexity: ?
-    Space Complexity: ?
+    Time Complexity: O(n)
+    Space Complexity: O(1)
     */
     delete(value) {
-        throw new Error('This method has\'t been implemented yet!')
+        (this.#head?.value === value)?(this.#head = this.#head.next):false;
```

Comment:
This line looks a little awkward with the assignment in the middle of the ternary.